staging/src/k8s.io/client-go/tools/remotecommand/remotecommand.go

```diff
@@ -43,11 +45,17 @@ type StreamOptions struct {
 
 // Executor is an interface for transporting shell-style streams.
 type Executor interface {
-	// Stream initiates the transport of the standard shell streams. It will transport any
-	// non-nil stream to a remote system, and return an error if a problem occurs. If tty
-	// is set, the stderr stream is not used (raw TTY manages stdout and stderr over the
-	// stdout stream).
+	// Deprecated: use StreamWithContext instead to avoid possible resource leaks.
+	// See https://github.com/kubernetes/kubernetes/pull/103177 for details.
 	Stream(options StreamOptions) error
+
+	// StreamWithContext initiates the transport of the standard shell streams. It will
+	// transport any non-nil stream to a remote system, and return an error if a problem
+	// occurs. If tty is set, the stderr stream is not used (raw TTY manages stdout and
+	// stderr over the stdout stream).
+	// The context controls the entire lifetime of stream execution including negotiating
```

Comment:
The negotiation isn't covered yet, right?

Reply:
Yep.

Can you remove the "negotiation" here and ask the author of #107605 to add it back? We can't guarantee that this PR and #107605 will be merged atomically.

Reply:
Just removed the description about the "negotiating and streaming". I'll comment on #107605 to suggest the revisal later.